openmls/src/group/core_group/test_proposals.rs

```diff
@@ -663,3 +636,145 @@ fn test_group_context_extension_proposal(
             .expect("Error exporting secret.")
     )
 }
+
+#[apply(ciphersuites_and_providers)]
+fn custom_proposals(ciphersuite: Ciphersuite, provider: &impl OpenMlsProvider) {
+    // Basic group setup.
+    let group_aad = b"Alice's test group";
+    let framing_parameters = FramingParameters::new(group_aad, WireFormat::PublicMessage);
+
+    let (alice_credential, _, alice_signer, _alice_pk) =
+        setup_client("Alice", ciphersuite, provider);
+    let (bob_credential_with_key, _bob_key_package_bundle, bob_signer, _) =
+        setup_client("Bob", ciphersuite, provider);
+
+    // Create a custom proposal type
+    let custom_proposal_id = 0xFF00;
+    let custom_proposal_type = ProposalType::Other(custom_proposal_id);
+    let custom_proposal_capabilities =
+        Capabilities::new(None, None, None, Some(&[custom_proposal_type]), None);
+    // Create Bob's key package with support for the custom proposal type
+    let bob_key_package = KeyPackageBuilder::new()
+        .leaf_node_capabilities(custom_proposal_capabilities.clone())
+        .build(ciphersuite, provider, &bob_signer, bob_credential_with_key)
+        .unwrap();
+
+    // Create group with custom proposal as required capability
+    let mut alice_group = CoreGroup::builder(
```

Comment:
Since the `CoreGroup` will disappear soon, we shouldn't write new tests for it and instead base them on `MlsGroup`.